### PR TITLE
{lib}[fosscuda/2018b] TensorRT v4.0.1.6

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
This easyconfig requires ~~https://github.com/easybuilders/easybuild-easyblocks/pull/1627~~

### CODE REVIEW

File: easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name = 'TensorRT'
+version = '4.0.1.6'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://developer.nvidia.com/tensorrt'
+description = """NVIDIA TensorRT is a platform for high-performance deep learning inference"""
+
+sources = ['%(name)s-%(version)s.Ubuntu-16.04.4.x86_64-gnu.cuda-9.2.cudnn7.1.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren Maybe mention that downloading manually via https://developer.nvidia.com/nvidia-tensorrt-4x-download is required in a comment?

File: easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+checksums = ['fc5fd2ba86225ed62e60870cc24896107770a59829363fcb49229c82bb1f4fb5']
+
+toolchain = {'name': 'fosscuda', 'version': '2018b'}
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren This should move up above `sources` to adher to our common easyconfigs "style"

File: easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ('protobuf', '3.6.1'),
+]
+
+dependencies = [
+    ('Python', '2.7.15'),
+    ('cuDNN', '7.1.4.18'),
+]
+
+extract_sources = True
+keepsymlinks = True
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren No longer needed thanks to custom easyblock for `TensorRT`?

File: easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+exts_list = [
+    ('protobuf-python', '3.6.1', {
+        'modulename': 'google.protobuf',
+        'source_tmpl': 'protobuf-python-%(version)s.tar.gz',
+        'source_urls': ['https://github.com/google/protobuf/releases/download/v%(version)s/'],
+        'start_dir': 'python',
+        'use_pip': False,
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren Why not use `pip` here, any specific reason?

Comment (akesandgren) [reply easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: Copied from the tensorflow easyconfig, so just following your lead :-)

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/30c82c2bef49f490dc537c49452f3cac for a full test report.

File: easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb (hunk 4 above)
Comment (boegel) [reply easybuild/easyconfigs/t/TensorRT/TensorRT-4.0.1.6-fosscuda-2018b-Python-2.7.15.eb]: Checked it again, there's apparently a good reason for it, it fails with an error when you try to use `pip`:

```
Can't find required file: ../src/google/protobuf/descriptor.proto
```

**Comment by akesandgren** [comment]
Kicking travis

**Comment by akesandgren** [comment]
Travis kicked

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3103.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/2417b4bc74900b8f8d8dbd99584e7f23 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2591.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1d32347573a8f852825b3242ca1bffbb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2065.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ecade570138f151d404507d216d119aa for a full test report.

**Comment by boegel** [comment]
Going in, thanks @akesandgren!

**Pull request merged**